CMSgov/mint-app
# Easi 2526/plan favorites user table

# EASI-2526

## Changes and Description

- Migration to change the data types that use EUAID in plan_favorite to by a foreign key reference to the user account table
- Update IBaseStruct to be able to accept `baseStructUserTable` 
Update unit tests so each test get's the user account each time.



## How to test this change
1.  This is testable from the front end. 
      a. Test that a favorite is added when you add a collaborator
             i. Add SF13 as a collab
             ii. Log in as SF13, and confirm that the plan is favorited for you
       b. Click and add a favorite for your account manually
       
2. Verify Unit tests

If you want to test the data migration, I recommend first checking out main, and adding test data as different user accounts, so there is a variety of different users in the plan favorite table. At that point, you can copy and paste the data migration from this branch into a SQL tool. Verify that the script ran correctly and all plan favorites now reference the user account table.


## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
